src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -358,10 +346,11 @@ private string CreateDetachScript(DetachDatabaseRequestParams detachParams, stri
         private CDataContainer CreateDatabaseDataContainer(string connectionUri, string? objectURN, bool isNewDatabase, string? databaseName)
         {
             ConnectionInfo connectionInfo = this.GetConnectionInfo(connectionUri);
-            if (!isNewDatabase && !string.IsNullOrEmpty(databaseName))
-            {
-                connectionInfo.ConnectionDetails.DatabaseName = databaseName;
-            }
+            // Update the database name in case the associated connection with the URI is for a different database like master.
+            // This also ensures we'll switch back to the original database name for this connection in subsequent calls so that it
+            // doesn't cause problems if we delete a database and this connection is still associated with that dropped database.
+            connectionInfo.ConnectionDetails.DatabaseName = databaseName ?? "master";
```

Comment:
I was running into issues where trying to open the New Database dialog would fail after Scripting a Detach Database operation and then dropping that same database. This is because we pass the ConnectionInfo by reference here, so that Detach operation would update the database name here. After dropping the database, trying to re-use the same connection would fail due to the associated database being dropped and the actual backing connection now getting cleaned up after each operation.

Reply:
@ssreerama any concerns about this change? I tested updates with the database properties dialog and didn't see any issues.